cmd/frontend/graphqlbackend/search.go

```diff
@@ -64,14 +67,80 @@ type SearchImplementer interface {
 	Stats(context.Context) (*searchResultsStats, error)
 }
 
+func alertForParseError(err error, queryString string) *searchAlert {
+	switch e := err.(type) {
+	case *syntax.ParseError:
+		return &searchAlert{
+			prometheusType:  "parse_syntax_error",
+			title:           capFirst(e.Msg),
+			description:     "Quoting the query may help if you want a literal match.",
+			proposedQueries: proposedQuotedQueries(queryString),
+		}
+	}
+	return &searchAlert{
+		prometheusType: "parse_syntax_error",
+		title:          "Query Parse Error",
+		description:    capFirst(err.Error()),
+	}
+}
+
+func alertForTypecheckError(err error, queryString string) *searchAlert {
+	switch e := err.(type) {
+	case *querytypes.TypeError:
+		switch e := e.Err.(type) {
+		case *rxsyntax.Error:
+			return &searchAlert{
+				prometheusType:  "typecheck_regex_syntax_error",
+				title:           capFirst(e.Error()),
+				description:     "Quoting the query may help if you want a literal match instead of a regular expression match.",
+				proposedQueries: proposedQuotedQueries(queryString),
+			}
+		}
+	}
+	return &searchAlert{
+		prometheusType: "typecheck_error",
+		title:          "Query Typecheck Error",
+		description:    capFirst(err.Error()),
+	}
+}
+
+// processQuery runs the processing pipeline that parses, type checks, and
+// validates search queries. Errors are converted to descriptive alerts or
+// suggestions that surface in the client.
```

Comment:
```suggestion
// suggestions that surface to the client.
```

Frankly, I don't know which one is correct in this content :)

Reply:
Probably 'are sent to' is clearer